crates/backend/src/services/frontend.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module contains the service which serves the backend as static files.
+
+use actix_files as fs;
+
+pub fn route_frontend(at: &str, dir: &str) -> actix_files::Files {
+    fs::Files::new(at, dir).index_file("index.html")
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use actix_web::{
+        http,
+        test,
+        App,
+    };
+    use std::{
+        env,
+        fs::File,
+        io::prelude::*,
+    };
+
+    #[actix_rt::test]
```

Comment:
Generally, it might be usefull to structure tests with a Given When Then structure: https://martinfowler.com/bliki/GivenWhenThen.html. But we can leave this for a later refactoring.